internal/cache/backend.go

```diff
@@ -43,14 +43,27 @@ func (b *Backend) Remove(ctx context.Context, h restic.Handle) error {
 	return b.Cache.Remove(h)
 }
 
-var autoCacheTypes = map[restic.FileType]struct{}{
-	restic.IndexFile:    {},
-	restic.SnapshotFile: {},
+func autoCacheTypes(ft restic.FileType, bt restic.BlobType) bool {
+	if ft == restic.IndexFile || ft == restic.SnapshotFile ||
+		(ft == restic.DataFile && bt != restic.DataBlob) {
```

Comment:
Distinguishing between mixed and tree-only packs is still a bit relevant for repositories created using old restic versions. For these the new behavior would amount to caching the whole backup repository, which seems rather problematic to me.